ndscan/fitting/fitting.py

```diff
@@ -0,0 +1,269 @@
+import numpy as np
+import functools
+import collections
+from typing import Dict, List, Optional, Tuple, Union
+from scipy import optimize
+
+
+# TODO: type annotations! (What's the best way of annotating np arrays?)
+# TODO: docstring formatting for sphinx
+class FitBase:
+    def __init__(
+        self,
+        x=None,
+        y=None,
+        y_err=None,
+        param_bounds: Optional[Dict[str, Tuple[float, float]]] = None,
+        fixed_params: Optional[Dict[str, float]] = None,
+        initial_values: Optional[Dict[str, float]] = None,
+        scale_factors: Optional[Dict[str, float]] = None,
```

Comment:
We could allow fit classes to specify default scale factors. This isn't so useful for general purpose fits like the sinusoid but might be for more specialised fit classes. I haven't bothered to add it for an initial implementation as it's easy to add later.